src/ray/raylet/worker_pool.cc

```diff
@@ -198,30 +207,50 @@ pid_t WorkerPool::StartProcess(const std::vector<std::string> &worker_command_ar
     RAY_LOG(DEBUG) << stream.str();
   }
 
-  // Launch the process to create the worker.
-  pid_t pid = fork();
-
-  if (pid != 0) {
-    return pid;
+  pid_t pid;
+  int err;
+#ifdef _WIN32
```

Comment:
The logic here is complicated/different enough between the two cases that it might be nice to separate these into functions

Reply:
Ah okay, let me give it a shot.

Reply:
Oh, regarding your top comment -- unfortunately it's not possible to test yet; the codebase doesn't compile on Windows yet. I do plan to replace this with `Boost.Process` once I can get the ball rolling so this won't be here for very long. :)

Reply:
I separated them out. Let me know if this is better?

Reply:
@edoakes Does it look good for merging? :)

Reply:
Nice, much easier to read!